src/py21cmfast/src/ps.c

```diff
@@ -2217,27 +2252,125 @@ float GaussLegendreQuad_Nion_MINI(int Type, int n, float growthf, float M2, floa
         .LimitMass_Fesc = Mlim_Fesc_MINI
     };
 
-    if(delta2 > delta1){
-        return 1.;
+
+
+
+    if(delta2 > delta1*0.9999) {
+        result = 1.;
+        return result;
+    }
+
+
+    if(FAST_FCOLL_TABLES){ //JBM: Fast tables. Assume sharp Mturn, not exponential cutoff.
+
+
+      if(MassTurnover_upper <= MassTurnover){
+        return 1e-40; //in sharp cut it's zero
+      }
+
+      double LogMass=log(MassTurnover);
+      int MassBin = (int)floor( (LogMass - MinMass )*inv_mass_bin_width );
+      double MassBinLow = MinMass + mass_bin_width*(double)MassBin;
+      double sigmaM1 = Sigma_InterpTable[MassBin] + ( LogMass - MassBinLow )*( Sigma_InterpTable[MassBin+1] - Sigma_InterpTable[MassBin] )*inv_mass_bin_width;
+      nu_lower_limit = (delta1-delta2)*(delta1-delta2)/(sigmaM1*sigmaM1-sigma2*sigma2)/growthf/growthf;
```

Comment:
I'd re-write this due to redundant calculations throughout.

Consider;

double delta_arg = pow( (delta1 - delta2)/growthf , 2.);

then have; 

nu_lower_limit = delta_arg * (sigmaM1*sigmaM1 - sigma2*sigma2)

You can then repeat this for nu_higher_limit, nupivot1 and nupivot2.

Reply:
Very true, I've changed it in both the regular and the MINI versions of the calculation. Thanks!